File: ui/backend/schemas/audit_package.py
"""Pydantic schemas for Screen 6 audit-package builder
(Phase 5 · PR-5d · DEC-V61-018)."""

from __future__ import annotations

from typing import List, Optional

from pydantic import BaseModel, Field


class AuditPackageDownloadUrls(BaseModel):
    """Server-relative URLs the UI links to for downloads."""

    manifest_json: str = Field(..., description="JSON manifest alone (convenience)")
    bundle_zip: str = Field(..., description="Byte-reproducible zip bundle")
    bundle_html: str = Field(..., description="Human-readable semantic HTML")
    bundle_pdf: Optional[str] = Field(
        None,
        description=(
            "PDF render (weasyprint). None when native libs unavailable; "
            "see `pdf_error` for actionable install hint."
        ),
    )
    bundle_sig: str = Field(..., description="HMAC-SHA256 hex signature sidecar (.sig)")


class AuditPackageVvChecklistItem(BaseModel):
    """One FDA V&V40 credibility-evidence area + manifest-field mapping."""

    area: str
    description: str
    manifest_fields: List[str]


class AuditPackageBuildResponse(BaseModel):
    """Returned by POST /cases/{id}/runs/{rid}/audit-package/build."""

    bundle_id: str = Field(..., description="32-hex uuid4; opaque to the UI")
    manifest_id: str
    case_id: str
    run_id: str
    generated_at: str
    git_repo_commit_sha: Optional[str]
    comparator_verdict: Optional[str] = Field(
        None, description="PASS | FAIL | HAZARD | None (no comparator run recorded)"
    )
    pdf_available: bool
    pdf_error: Optional[str] = Field(
        None,
        description=(
            "Install hint when PDF render failed. Expect non-None when "
            "`pdf_available` is False."
        ),
    )
    downloads: AuditPackageDownloadUrls
    vv40_checklist: List[AuditPackageVvChecklistItem]
    signature_hex: str = Field(
        ...,
        description=(
            "64-char hex HMAC-SHA256. UI may display or hide — the sidecar "
            ".sig file is the canonical persistence."
        ),
    )

File: ui/frontend/src/types/audit_package.ts
// Phase 5 · Screen 6 — Audit Package Builder types.
// Mirrors ui/backend/schemas/audit_package.py.

export interface AuditPackageDownloadUrls {
  manifest_json: string;
  bundle_zip: string;
  bundle_html: string;
  bundle_pdf: string | null;
  bundle_sig: string;
}

// Renamed from AuditPackageVvChecklistItem per Codex PR-5d MEDIUM finding:
// the 8-row table is a product-specific summary, not a faithful rendering
// of the FDA/ASME V&V40 framework.
export interface AuditPackageEvidenceItem {
  area: string;
  description: string;
  manifest_fields: string[];
}

export interface AuditPackageBuildResponse {
  bundle_id: string;
  manifest_id: string;
  case_id: string;
  run_id: string;
  generated_at: string;
  git_repo_commit_sha: string | null;
  comparator_verdict: string | null;
  pdf_available: boolean;
  pdf_error: string | null;
  downloads: AuditPackageDownloadUrls;
  evidence_summary: AuditPackageEvidenceItem[];
  signature_hex: string;
}
